ada/extensions/src/libadalang-doc_utils.adb

```diff
@@ -136,12 +136,29 @@ package body Libadalang.Doc_Utils is
       --  Transform the doc vector into a string
       declare
          Last_Index : Natural := Doc_Vec.Last_Index;
+
+         First_Line : XString :=
+           (if Last_Index /= 0 then Doc_Vec.First_Element else Null_XString);
+
+         Offset     : Positive :=
+           First_Line.Length - First_Line.Trim.Length + 1;
+         --  Offset for the first non whitespace char on the first line. Will
+         --  be used as the offset to start every subsequent doc line.
```

Comment:
The previous code was taking the minimum indentation (and thus never failed) while here we take the indentation of the first line and raise an error if it’s not the minimum indentation. Given that this property is just for laldoc’s purposes, i.e. for doc that we write ourselves, I guess that’s fine, but I hope we won’t get libadalang-doc build failures because of poorly formatted comments in the future (the testsuite does not catches that for now).

Reply:
For the moment, the only use case of get_doc is to generate sphinx. With the old algorithm, you would get very weirdly formatted sphinx anyway if you used the old algorithm in a case this algorithm doesn't handle, and probably not what you wanted in the first place (like starting with an indented block)

Reply:
And actually the testsuite does have a case of unproperly formatted comment so I'm not sure what you mean ?